### DISCUSSION TITLE
What is the best option to keep commits low?

### DISCUSSION THREAD

**Discussion by Andre601** [lowlighter/metrics#885] [Q&A] [answered]
The goal is simple: Reduce amount of Commits the GitHub Action does for updating the README file.

The action as of right now creates 3 separate commits towards the main branch.
Checking the last update, the Commits seem to be as follows:

- 1 commit updating the Stargazers SVG used. ([Commit](https://github.com/Readme-Workflows/recent-activity/commit/76a9a3863a0dcc99264002a23f8b2f0924478f87))
- 2 empty commits with no changes whatsoever according to GitHub ([Commit 1](https://github.com/Readme-Workflows/recent-activity/commit/b92073fc17b98a4361d39e0043e5e8177b8b6d69), [Commit 2](https://github.com/Readme-Workflows/recent-activity/commit/a90268745f831c259da845c5f063170e5d51ccfb))

Perhaps the action does a small change in the other commits that GitHub simply isn't aware of/doesn't notice?

Either way, I would like to reduce the count, since we use webhooks for Discord activities, including commits, and this can and will flood the channel every so often.

I saw that there is a Pull request type option, but this wouldn't help too much with the issue at hand.

Are there any other options that we could try?

**Comment by lowlighter** [comment, marked as answer]
Maybe you can find some leads in https://github.com/lowlighter/metrics/discussions/471

But tldr;, here's a few solutions you could try:
- change [`committer_branch`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#committer_branch) and use an orphan branch instead unrelated to default branch
- use [`output_action: none`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#output_action) and manually handle what to do with result
- use [gists](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#using-gists) instead

But I think only the first solution may work currently for your use case, I'm not sure that `.cache` files inherits the required params to use the two later option

In addition, with the first solution it may be easier to avoid the hooks since you just need to ignore the dest branch instead, and it avoids polluting the default branch
